### DISCUSSION TITLE
MagickResourceLimitErrorException when using WebP define 'thread-level'

### DISCUSSION THREAD

**Discussion by Cyl18** [dlemstra/Magick.NET#779] [Help] [answered]
![image](https://user-images.githubusercontent.com/14993992/98447169-9c20c300-215d-11eb-952c-7775a225cd53.png)
Tried AnyCPU and OpenMP
using Q8 7.22.1

**Comment by dlemstra** [comment, marked as answer]
Do you have a small code sample that reproduces this issue?

**Reply by Cyl18** [reply]
> Do you have a small code sample that reproduces this issue?
@dlemstra 
```CSharp
public void Test()
{
    var sourceImage = new MagickImage("a.png");
    var webPMemoryStream = new MemoryStream();
    
    sourceImage.Write(webPMemoryStream, new WebPDefines());
}

public class WebPDefines : IWriteDefines
{
    public IEnumerable<IDefine> Defines { get; } = new[]
    {
        new MagickDefine(MagickFormat.WebP, "thread-level", "2"),
    };
    public MagickFormat Format { get; } = MagickFormat.WebP;
}
```

**Reply by dlemstra** [reply]
Turns out that value is actually a boolean:

```c
 int thread_level;       // If non-zero, try and use multi-threaded encoding.
```

You should use `1` instead of `2`.

**Reply by Cyl18** [reply]
uh, that's my problem. I thought it would be a thread count
i found this define in https://imagemagick.org/script/webp.php 
the document is not clear, haha, thx

**Reply by dlemstra** [reply]
Thanks for your remark about the documentation not being clear. Just pushed a commit to our website repository. That should be on the website some time tomorrow.
